controller/uninstall_controller.go

```diff
@@ -541,6 +591,43 @@ func (c *UninstallController) deleteInstanceManagers(instanceManagers map[string
 	return
 }
 
+func (c *UninstallController) deleteShareManagers(shareManagers map[string]*longhorn.ShareManager) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "Failed to delete share managers")
+	}()
+	for _, sm := range shareManagers {
+		log := getLoggerForShareManager(c.logger, sm)
+
+		timeout := metav1.NewTime(time.Now().Add(-gracePeriod))
+		if sm.DeletionTimestamp == nil {
+			if err = c.ds.DeleteShareManager(sm.Name); err != nil {
+				err = errors.Wrapf(err, "Failed to mark for deletion")
+				return
+			}
+			log.Info("Marked for deletion")
+		} else if sm.DeletionTimestamp.Before(&timeout) {
+			var pod *v1.Pod
+			pod, err = c.ds.GetPod(sm.Name)
+			if err != nil {
+				err = errors.Wrapf(err, "Could not get pod for share manager")
+			}
+			if pod != nil {
+				if err = c.ds.DeletePod(pod.Name); err != nil {
```

Comment:
I think you can just call delete without initial retrieval of the pod?
```
podName := types.GetShareManagerPodNameFromShareManagerName(sm.Name)`
c.ds.DeletePod(podName)
```

Reply:
changed